lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -394,4 +396,22 @@ public function getReflectionService()
         }
         return $this->reflectionService;
     }
+
+    /**
+     * Gets the real class name of a class name that could be a proxy.
+     *
+     * @param string $class
+     *
+     * @return string
+     */
+    private function getRealClass($class)
```

Comment:
Shouldn't this be part of a `doctrine/proxy` implementation?

Reply:
We don't really want doctrine/proxy, it's to be dropped soon.

Reply:
Gotcha 👍